instance/models/utils.py

```diff
@@ -591,10 +591,14 @@ def delete(self, key, **kwargs):
 
     def purge(self):
         """
-        Will simply removes all keys/prefixed-keys from Key-Value store.
+        Will remove the key/prefixed-key from Key-Value store.
```

Comment:
@pkulkark, this method is only called from one place in the code and that is specific to an instance. Since the configuration data for an instance is stored in a single k-v pair, I don't see a reason why we need the recursive deletion at all.

Before, we used a prefix like `ocim/instances/<id>/` and stored multiple k-v pairs under that. The recursive deletion would have been needed only in that case. What do you say?

There is an unused `remove_dict` method which already does this. So we can move its logic to the `purge()` method, remove the `remove_dict()` method and the associated unit test.

CC @clemente, @pcockwell

Reply:
I don't follow very well how prefixes are used; there seem to be many equivalent ways to operate with keys. I think our current case doesn't need recursive, but imagine that in the future we switch
- from `ocim/instances/123`
- to `ocim/instances/123/encrypted_data`, `ocim/instances/123/plaintext_data`

Then by asking „please delete 123 recursively“, the difference between recursive and non-recursive is relevant (I guess the 2nd case would delete more).
If we just store 1 key called 123, we don't need to care in practice (though the code may be untested in the cases we don't use).
Anyway, I'm not familiar with how we use prefixes and I don't care about the implementation. Just make sure that the tests cover the dangerous parts…

Reply:
@clemente, afaik, we use prefixes in two contexts - the first to restrict all operations to the `ocim/instances` prefix. The second is to delete all the keys under an instance's prefix - `ocim/instances/1234/*`. We migrated away from the second a while ago.

We only store one key per instance now and so there is no need to recursively delete anything at all from the only code that calls `.purge()`, which is in the context of an instance.

Reply:
@lgp171188 From the unit tests it looked like the expected behaviour is to delete all key-value pairs if no prefix is sent to the purge method. Should that be kept as is?

Reply:
I've removed the `remove_dict()` method and the corresponding test case.

Reply:
> @lgp171188 From the unit tests it looked like the expected behaviour is to delete all key-value pairs if no prefix is sent to the purge method. Should that be kept as is?

@pkulkark, I think this is not needed any more because:
* We don't store the instance configuration in multiple keys under a prefix.
* Other than the previous scenario, the only other scenario where deleting all key-pairs will be needed is if we have to delete all the instance data from consul. I'm sure we don't want that happening either.

So I think it should be okay to not expect or test the behaviour in question.

Reply:
Thanks @lgp171188. I've made the required updates.